contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -233,10 +233,14 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20Upgradeable, ReentrancyGuardUpg
         uint256 totalSupply = totalSupply();
         for (uint256 i = 0; i < tokenList.length; i++) {
             address token = tokenList[i];
-            _clearPendingRewards(from, token);
-            _clearPendingRewards(to, token);
-            $.claimerInfo[token][to] = ($.poolInfo[token] * (balanceOf(to) + amount)) / totalSupply;
-            $.claimerInfo[token][from] = ($.poolInfo[token] * (balanceOf(from) - amount)) / totalSupply;
+            uint256 pendingFrom = _claimableRevenue(from, token);
+            uint256 pendingTo = _claimableRevenue(to, token);
+            $.claimerInfo[token][to] =
+                int256(($.poolInfo[token] * (balanceOf(to) + amount)) / totalSupply) -
+                int256(pendingTo);
+            $.claimerInfo[token][from] =
```

Comment:
$.claimerInfo[tokenList[i]][from] =
                int256(($.poolInfo[tokenList[i]] * (balanceOfFrom - amount)) / totalSupply) -
                int256(pendingFrom);

Will it revert on royalty token transfer if the user has not claimed yet?

Because,
($.poolInfo[tokenList[i]] * (balanceOfFrom - amount)) / totalSupply) < pendingFrom

Thinking of the DEX contract that holds the royalty tokens for swapping - would it grind to a halt? Or am I missing something?

Reply:
It would out revert, since the value type of `claimerInfo` is `int256` which hold negative value.
Such case that transfer RTs while still have pending revenue, has also been covered in test case `test_IpRoyaltyVault_payRev_then_transferRTs`:
https://github.com/storyprotocol/protocol-core-v1/pull/291/files#diff-7b1113b9f56f60a900b64a18ab4e5962438cba8524418eda505d359e8c83c0cfR533